### PR TITLE
whitelist code length adherence to documentation

### PR DESCRIPTION
**Pull request by braunfuss**
Right now for the whitelist only NET.STA.LOC, or NET.STA.LOC.CHA are supported.  STA and NET.STA are however documented. Special discrimination is given to the whitelist but not the blacklist in this regard.

### CODE REVIEW

File: src/dataset.py [hunk 1]
```diff
@@ -314,7 +314,12 @@ def is_whitelisted(self, obj):
             return True
 
         nsl = self.get_nsl(obj)
+
         try:
+            if nsl[0:2] in self.whitelist_nsl:
+                return True
+            if nsl[0:1] in self.whitelist_nsl:
```
[the remark this answers is not in the document]
Comment (braunfuss) [reply src/dataset.py]: Correct and fixed. We had a test case with a whitelisted NET, so it slipped by. The option to blacklist/whitelist by NET could be neat anyway?

**Comment by braunfuss** [comment]
450d8d6 works also fine, however we might want to add also the Network code only as whitelist/blacklist options?

**Comment by emolch** [comment]
This would be ambiguous with station only.

**Comment by emolch** [comment]
A better way would have been to allow for patterns. `NET.*.*.*, *.STA.*.*, ... *.*.*.BH?`

**Comment by emolch** [comment]
Maybe that would be a better way: translate single names like e.g. `STA` into `*.STA.*.*` and then have simple logic in the `is_whitelisted`, `is_blacklisted` functions. But also this approach will need a fallback for the case where an object providing only nsl is matched...

**Comment by braunfuss** [comment]
Could be done similar to the snufflers hide routine perhaps.

**Pull request merged**